avocado/core/test.py

```diff
@@ -615,6 +628,14 @@ def _check_reference_stderr(self):
                 os.path.isfile(self._expected_stderr_file)):
             expected = genio.read_file(self._expected_stderr_file)
             actual = genio.read_file(self._stderr_file)
+
+            diff = unified_diff(expected.splitlines(), actual.splitlines(),
+                                fromfile=self._expected_stderr_file,
+                                tofile=self._stderr_file)
+            with open(self._stderr_diff_file, 'w') as file_obj:
+                for line in diff:
+                    file_obj.write(line.strip()+'\n')
+
             msg = ('Actual test sdterr differs from expected one:\n'
```

Comment:
I'm not sure I understand the card properly https://trello.com/c/CXDP1khV/1106-check-output-functionality-should-also-print-a-diff-expected-found but at least the heading says "print a diff" so the way I understand the card is it should log the diff, store the actual output and the diff, where the diff should be stored instead of the previously logged `expected` file.

Reply:
@clebergnu wrote the card, maybe he can detail the expected implementation. @clebergnu?

Reply:
Just recording a cross reference that answers this:

https://trello.com/c/CXDP1khV/1106-check-output-functionality-should-also-print-a-diff-expected-found#comment-59e0be3d6a52a43ef49f305d